prov/shm/src/smr_msg.c

```diff
@@ -183,33 +188,39 @@ static ssize_t smr_generic_sendmsg(struct smr_ep *ep, const struct iovec *iov,
 		goto unlock_cq;
 	}
 
+	iface = ofi_get_mr_device_iface(&ep->util_ep.domain->mr_map, desc, &device);
+
 	total_len = ofi_total_iov_len(iov, iov_count);
 
 	cmd = ofi_cirque_tail(smr_cmd_queue(peer_smr));
 	smr_generic_format(cmd, peer_id, op, tag, data, op_flags);
 
 	if (total_len <= SMR_MSG_DATA_LEN && !(op_flags & FI_DELIVERY_COMPLETE)) {
-		smr_format_inline(cmd, iov, iov_count);
+		smr_format_inline(cmd, iface, device, iov, iov_count);
 	} else if (total_len <= SMR_INJECT_SIZE &&
 		   !(op_flags & FI_DELIVERY_COMPLETE)) {
 		tx_buf = smr_freestack_pop(smr_inject_pool(peer_smr));
-		smr_format_inject(cmd, iov, iov_count, peer_smr, tx_buf);
+		smr_format_inject(cmd, iface, device, iov, iov_count, peer_smr, tx_buf);
```

Comment:
Since inject is not supported with FI_HMEM, should iface be hardcoded to FI_HMEM_SYSTEM and device to 0? This aligns with the other smr_format_inject() change.

Reply:
Nevermind. I see that this format is used elsewhere.